apis/logging/v1/cluster_log_forwarder.go

```diff
@@ -196,19 +197,42 @@ func (output *OutputSpec) GetMaxRecordsPerSecond() int64 {
 	return output.Limit.MaxRecordsPerSecond
 }
 
-func IsAuditHttpReceiver(input *InputSpec) bool {
-	return input.Receiver != nil &&
-		input.Receiver.HTTP != nil &&
-		input.Receiver.Type == ReceiverTypeHttp &&
-		input.Receiver.HTTP.Format == FormatKubeAPIAudit
+func (receiver *ReceiverSpec) IsAuditHttpReceiver() bool {
+	return receiver != nil &&
+		receiver.Type == ReceiverTypeHttp &&
+		receiver.GetHTTPFormat() == FormatKubeAPIAudit
 }
 
-func IsHttpReceiver(input *InputSpec) bool {
-	return input.Receiver != nil &&
-		input.Receiver.Type == ReceiverTypeHttp
+func (receiver *ReceiverSpec) IsHttpReceiver() bool {
+	return receiver != nil &&
+		receiver.Type == ReceiverTypeHttp
 }
 
-func IsSyslogReceiver(input *InputSpec) bool {
-	return input.Receiver != nil &&
-		input.Receiver.Type == ReceiverTypeSyslog
+func (receiver *ReceiverSpec) IsSyslogReceiver() bool {
```

Comment:
Probably a nit but do we really need these or the port functions? We already have type which can be used to get the port from the correct subfield